build.tex

```diff
@@ -1,5 +1,54 @@
+
+Support packages from the Science Quality and Reliability Engineering group, use industry standard packaging systems based on the language in the package, with Python packages being distributed through PyPI and dependencies managed in the standard Python manner.
+
+\subsubsection{Building from git}
+
+With all the prerequisites installed we needed a way to retrieve all the source code from GitHub and to work out which packages should be built and in what order.
+We have implemented this by writing two support packages that use EUPS for dependency management and \texttt{eupspkg} for builds.
+The \texttt{lsstsw} package was originally written purely to support continuous integration services but is now the default choice for developers who wish to build the most up to date version of the software from source.
+There is a shell script to install the local build system by installing Python and EUPS and there is a Python program for doing a full build.
+This \texttt{rebuild} program is a wrapper around the \texttt{lsst\_build} package.
+There are two key phases to doing a build.
+In the first phase all the source is cloned from GitHub and the correct git reference is checked out based on the user request (usually you list the feature branch name that you are testing).
+We have a master list in a GitHub repository that maps product name to git repository and the command can be made to refresh this list every time it runs.
+Once the code has been cloned and the source repositories configured properly, the second phase is to build the code in dependency order and install it using EUPS tagged with the current build number.
+The build tag can then be used to setup the products from that specific build.
+
+\subsubsection{Building from a source distribution}
+
+Not everyone needs to be able to build the cutting edge version of the code and for those users we publish packages to an EUPS server and EUPS can download the packages associated with a public EUPS tag.
+This is usually a weekly, for example \texttt{w\_2018\_13}, or an official release, such as \texttt{v15\_0}.
+
+\subsubsection{Future Considerations}
+
+We have considered creating a git repository similar to the current \texttt{lsst\_distrib} metapackage, containing git submodules to the separate git repositories, and this is how we implement a single git repository of all the DM documentation.
+This would have the advantage of being able to check out specific versions of all the desired packages, and whilst the updates to the metapackage could be automated, this comes at the added complication of submodules.
+We are also investigating the adoption of true semantic versioning\cite{semver} for all our modules and conversion of the packaging to standard Python packages that can be \texttt{pip}-installable (see for example the discussion in Ref.~\citenum{2016SPIE.9913E..0GJ}) but this is not yet past a proof of concept stage.
```

Comment:
Re-adoption, actually; we did semver through at least 2011.

Reply:
can you give me a sentence saying why we stopped?